docs/compute-element/job-router-recipes.md

```diff
@@ -0,0 +1,762 @@
+Writing Routes For HTCondor-CE
+==============================
+
+The [JobRouter](http://research.cs.wisc.edu/htcondor/manual/v8.6/5_4HTCondor_Job.html) is at the heart of HTCondor-CE and allows admins to transform and direct jobs to specific batch systems. Customizations are made in the form of job routes where each route corresponds to a separate job transformation: If an incoming job matches a job route's requirements, the route creates a transformed job (referred to as the 'routed job') that is then submitted to the batch system. The CE package comes with default routes located in `/etc/condor-ce/config.d/02-ce-*.conf` that provide enough basic functionality for a small site.
+
+If you have needs beyond delegating all incoming jobs to your batch system as they are, this document provides examples of common job routes and job route problems.
+
+Quirks and Pitfalls
+-------------------
+
+-   The JobRouter matches jobs to routes in a round-robin fashion (for HTCondor versions < 8.7.1). This means that if a job can match to multiple routes, it can be routed by any of them! So when writing job routes, make sure that they are exclusive to each other and that your jobs can only match to a single route.
+-   If a value is set in [JOB\_ROUTER\_DEFAULTS](#job59router59defaults) with `eval_set_<variable>`, override it by using `eval_set_<variable>` in the `JOB_ROUTER_ENTRIES`. Do this at your own risk as it may cause the CE to break.
+-   Make sure to run `condor_ce_reconfig` after changing your routes, otherwise they will not take effect.
+-   Before the last square bracket, make sure all lines end in a line continuation character (backslash). You can inspect the syntax of your routes with `condor_ce_config_val JOB_ROUTER_ENTRIES` to see if HTCondor-CE has ingested them properly.
+-   Do **not** set the job environment through the JobRouter. Instead, add any changes to the `[Local Settings]` section in `/etc/osg/config.d/40-localsettings.ini` and run osg-configure, as documented [here](../other/configuration-with-osg-configure).
+-   HTCondor batch system only: Local universe jobs are excluded from any routing.
+
+How Job Routes are Constructed
+------------------------------
+
+Each job route’s [ClassAd](http://research.cs.wisc.edu/htcondor/manual/v8.6/4_1HTCondor_s_ClassAd.html) is constructed by combining each entry from the `JOB_ROUTER_ENTRIES` with the `JOB_ROUTER_DEFAULTS`. Attributes that are [set_](#setting-attributes) in `JOB_ROUTER_ENTRIES` will override those [set_](#setting-attributes) in `JOB_ROUTER_DEFAULTS`
+
+### JOB\_ROUTER\_ENTRIES
```

Comment:
Underscores here and in the following headers don't need to be backslashed (and it messes up links).